### PR TITLE
[SPARK-36254][INFRA][PYTHON] Install mlflow/sklearn in Github Actions CI

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes adding a Python package, `mlflow` and `sklearn` to enable the MLflow test in pandas API on Spark.

### Why are the changes needed?

To enable the MLflow test in pandas API on Spark.

### Does this PR introduce _any_ user-facing change?

No, it's test-only

### How was this patch tested?

Manually test on local, with `python/run-tests --testnames pyspark.pandas.mlflow`.

### CODE REVIEW

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -157,7 +157,7 @@ jobs:
     - name: Install Python packages (Python 3.8)
       if: (contains(matrix.modules, 'sql') && !contains(matrix.modules, 'sql-'))
       run: |
-        python3.8 -m pip install numpy 'pyarrow<5.0.0' pandas scipy xmlrunner
+        python3.8 -m pip install numpy 'pyarrow<5.0.0' pandas scipy xmlrunner 'mlflow>=1.0' sklearn
```
Comment (HyukjinKwon) [inline .github/workflows/build_and_test.yml:160]: can you install it in pyspark job?

**Review by HyukjinKwon — COMMENTED** [review]
please also update https://github.com/apache/spark/blob/master/dev/requirements.txt

File: .github/workflows/build_and_test.yml (hunk 1 above)
Comment (dongjoon-hyun) [reply .github/workflows/build_and_test.yml:160]: +1 for @HyukjinKwon 's comment.

File: .github/workflows/build_and_test.yml [hunk 2]
```diff
@@ -242,8 +242,9 @@ jobs:
         key: pyspark-coursier-${{ hashFiles('**/pom.xml', '**/plugins.sbt') }}
         restore-keys: |
           pyspark-coursier-
-    - name: List Python packages (Python 3.9)
+    - name: Install Python packages (Python 3.9)
```
Comment (HyukjinKwon) [inline .github/workflows/build_and_test.yml:245]: Let's revert this change.

File: .github/workflows/build_and_test.yml [hunk 3]
```diff
@@ -242,8 +242,9 @@ jobs:
...
       run: |
+        python3.9 -m pip install 'mlflow>=1.0' sklearn
```
Comment (HyukjinKwon) [inline .github/workflows/build_and_test.yml:247]: Let's file a JIRA for the image to have `'mlflow>=1.0' sklearn`. Once the image has both packages, we can remove here. Add a comment `# TODO(SPARK-XXXXX): Install blah blah`.

Comment (itholic) [reply .github/workflows/build_and_test.yml:248]: Sounds good. Just created a [ticket](https://issues.apache.org/jira/browse/SPARK-36345) and added comment.

File: .github/workflows/build_and_test.yml [hunk 4]
```diff
@@ -244,6 +244,8 @@ jobs:
           pyspark-coursier-
     - name: List Python packages (Python 3.9)
       run: |
+        # TODO(SPARK-36345): We can remove the pip install here, by creating the image that already has both package installed.
```
Comment (HyukjinKwon) [inline .github/workflows/build_and_test.yml:247]:
```suggestion
        # TODO(SPARK-36345): Install mlflow>=1.0 and sklearn in Python 3.9 of the base image
```

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
@itholic seems GA is stuck. Can you rebase to retrigger the GA build?

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Yep, as @HyukjinKwon wrote, we should see GitHub Action result on this PR.

File: .github/workflows/build_and_test.yml [hunk 5]
```diff
@@ -244,6 +244,8 @@ jobs:
           pyspark-coursier-
     - name: List Python packages (Python 3.9)
       run: |
+        # TODO(SPARK-36345): Install mlflow>=1.0 and sklearn in Python 3.9 of the base image
+        python3.9 -m pip install 'mlflow>=1.0' sklearn
```
Comment (HyukjinKwon) [inline .github/workflows/build_and_test.yml:248]: sorry actually let's move this right before `export PATH=$PATH:$HOME/miniconda/bin` at `Run tests`.

Comment (itholic) [reply .github/workflows/build_and_test.yml]: Sure, just moved!

**Review by dongjoon-hyun — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Merged to master/3.2.

File: .github/workflows/build_and_test.yml [hunk 6]
```diff
@@ -252,6 +252,8 @@ jobs:
     # Run the tests.
     - name: Run tests
       run: |
+        # TODO(SPARK-36345): Install mlflow>=1.0 and sklearn in Python 3.9 of the base image
```
Comment (dongjoon-hyun) [inline .github/workflows/build_and_test.yml:255]: I'll proceed SPARK-36345.

**Comment by Yikun** [comment]
```python
File "/__w/spark/spark/python/pyspark/pandas/mlflow.py", line 142, in pyspark.pandas.mlflow.load_model
Failed example:
    from mlflow.tracking import MlflowClient, set_tracking_uri
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib/python3.9/doctest.py", line 1336, in __run
        exec(compile(example.source, filename, "single",
      File "<doctest pyspark.pandas.mlflow.load_model[0]>", line 1, in <module>
        from mlflow.tracking import MlflowClient, set_tracking_uri
    ModuleNotFoundError: No module named 'mlflow'
**********************************************************************
```

I'm not sure relate or not: the doctest of pyspark.pandas.mlflow raise the `ModuleNotFoundError`.

[1] https://github.com/Yikun/spark/runs/3199633887?check_suite_focus=true

**Comment by HyukjinKwon** [comment]
I think that's because it checks out to the master, and https://github.com/apache/spark/pull/33567/files#diff-bf4df96aadfad9579e20eccf9a30095927402142b79fe53ea5068cd290ab7f17L232-L238 was removed. That would happen when the forked repository's head is behind the Apache Spark's master branch

**Comment by HyukjinKwon** [comment]
probably we should partially revert so the tests run without installing mlflow, etc. let me make a PR.

**Comment by Yikun** [comment]
> I think that's because it checks out to the master, and https://github.com/apache/spark/pull/33567/files#diff-bf4df96aadfad9579e20eccf9a30095927402142b79fe53ea5068cd290ab7f17L232-L238 was removed

@HyukjinKwon Yep, the CI is passed after I rebase the commits to upstream/master.
